packages/jupyterlab-lsp/src/adapters/jupyterlab/jl_adapter.ts

```
@@ -32,6 +36,37 @@ import {
   ISocketConnectionOptions
 } from '../../connection_manager';
 import { Rename } from '../codemirror/features/rename';
+import { LSPExtension } from '../../index';
+import { ISettingRegistry } from '@jupyterlab/settingregistry';
+
+class LabFeatureSettings implements IFeatureSettings {
+  private key = 'features';
+
+  constructor(
+    private extension: LSPExtension,
+    protected feature_name: string
+  ) {}
+
+  protected get lab_settings(): ISettingRegistry.ISettings {
+    return this.extension.settings;
+  }
+
+  protected get_obj(): any {
+    return (this.lab_settings.get(this.key).composite as any)[
+      this.feature_name.toLowerCase()
+    ];
+  }
+
+  get(setting: string) {
+    return this.get_obj()[setting];
+  }
+
+  set(setting: string, value: any) {
+    let obj = this.get_obj();
+    obj[setting] = value;
+    this.lab_settings.set(this.key, obj).catch(console.warn);
```

it seems it would be reasonable to make this return a `Promise<void>` instead of swallowing the error... at some point, we should figure out how to actually validate these...